exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Otel2PrometheusConverter.java

```diff
@@ -400,29 +403,42 @@ private Exemplars convertDoubleExemplars(List<DoubleExemplarData> exemplars) {
     return Exemplars.of(result);
   }
 
+  @Nullable
   private Exemplar convertExemplar(double value, ExemplarData exemplar) {
     SpanContext spanContext = exemplar.getSpanContext();
+    Labels labels = Labels.EMPTY;
     if (spanContext.isValid()) {
-      return new Exemplar(
-          value,
+      labels =
           convertAttributes(
-              null, // resource attributes are only copied for point's attributes
-              null, // scope attributes are only needed for point's attributes
+              null,
+              null,
               exemplar.getFilteredAttributes(),
               "trace_id",
               spanContext.getTraceId(),
               "span_id",
-              spanContext.getSpanId()),
-          exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+              spanContext.getSpanId());
     } else {
-      return new Exemplar(
-          value,
-          convertAttributes(
-              null, // resource attributes are only copied for point's attributes
-              null, // scope attributes are only needed for point's attributes
-              exemplar.getFilteredAttributes()),
-          exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+      labels = convertAttributes(null, null, exemplar.getFilteredAttributes());
+    }
+    int runes = getRunes(labels);
+    if (runes > EXEMPLAR_MAX_RUNES) {
+      THROTTLING_LOGGER.log(
+          Level.WARNING,
+          "exemplar labels have " + runes + " runes, exceeding the limit of " + EXEMPLAR_MAX_RUNES);
+      return null;
+    }
+    return new Exemplar(value, labels, exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+  }
+
+  private static int getRunes(Labels labels) {
```

Comment:
I think `rune` is a go term. In java I believe it is more appropriate to call them code points. I'm not an expert in unicode terminology so could get this wrong. As far as I can tell this method currently adds up a bunch of string lengths. In java string length is expressed in utf-16 code units. From a quick google search I got the impression that a go rune corresponds to a unicode code point so I'd guess that this method will end up with a higher rune count than what go would in case the string contains surrogate pairs. Perhaps you should use `codePointCount` instead of `length`? Ideally you should verify that your method and the go method do the same thing.

Reply:
hi @laurit,
any further feedback ?

Reply:
Looks fine to me. You should pay attention to https://github.com/open-telemetry/opentelemetry-java/issues/6770#issuecomment-2414097107 Perhaps this issue will be solved by changing the prometheus java client instead.